### PR TITLE
Move logic outside AST

### PR DESCRIPTION
**Pull request by ishche**

### CODE REVIEW

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/DeclarativeProcedureSectionRegister.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import java.util.function.BiConsumer;
+
+/** DeclarativeProcedureSectionNode processor */
+public class DeclarativeProcedureSectionRegister
+    implements BiConsumer<DeclarativeProcedureSectionNode, ProcessingContext> {
+  @Override
+  public void accept(DeclarativeProcedureSectionNode node, ProcessingContext processingContext) {
+    Optional<ProgramNode> programOpt = node.getProgram();
+    if (!programOpt.isPresent()) {
+      // TODO: error?
```
Comment (lxpdd) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/DeclarativeProcedureSectionRegister.java:32]: please assign this TODO to some issue or @username

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/SectionNameRegister.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import java.util.function.BiConsumer;
+
+import static org.eclipse.lsp.cobol.core.model.tree.NodeType.PROCEDURE_SECTION;
+
+/** SectionNameNode processor */
+public class SectionNameRegister implements BiConsumer<SectionNameNode, ProcessingContext> {
+  @Override
+  public void accept(SectionNameNode node, ProcessingContext ctx) {
+    if (node.getParent().getNodeType() != PROCEDURE_SECTION) {
+      // TODO: register usage
```
Comment (lxpdd) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/SectionNameRegister.java:31]: again TODO without detailed explanation

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/statements/SetToBooleanStatement.java [hunk 3]
```diff
@@ -15,33 +15,21 @@
...
 
 /** This class implements the logic for SET TO TRUE/FALSE statement. */
 @EqualsAndHashCode(callSuper = true)
+@Getter
 public class SetToBooleanStatement extends StatementNode {
-  protected static final String INVALID_RECEIVING_FIELD_TEMPLATE =
-      "statements.invalidReceivingField";
-  private static final List<VariableType> ALLOWED_TYPES =
-      ImmutableList.of(VariableType.CONDITION_DATA_NAME);
   List<Node> receivingFields;
```
Comment (lxpdd) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/statements/SetToBooleanStatement.java:29]: make this field private

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/statements/SetToOnOffStatement.java [hunk 4]
```diff
@@ -15,33 +15,21 @@
...
 
 /** This class implements the logic for SET TO ON/OFF statement. */
 @EqualsAndHashCode(callSuper = true)
+@Getter
 public class SetToOnOffStatement extends StatementNode {
-  protected static final String INVALID_RECEIVING_FIELD_TEMPLATE =
-      "statements.invalidReceivingField";
-  private static final List<VariableType> ALLOWED_TYPES =
-      ImmutableList.of(VariableType.MNEMONIC_NAME);
   List<Node> receivingFields;
```
Comment (lxpdd) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/statements/SetToOnOffStatement.java:29]: private

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/statements/SetUpDownByStatement.java [hunk 5]
```diff
@@ -15,37 +15,17 @@
...
-  protected static final String INVALID_SENDING_FIELD_TEMPLATE = "statements.invalidSendingField";
-  private static final List<VariableType> RECEIVING_FIELD_TYPES =
-      ImmutableList.of(VariableType.INDEX_ITEM);
-  private static final List<EffectiveDataType> ALLOWED_SENDING_FIELDS_TYPE =
-      ImmutableList.of(EffectiveDataType.INTEGER, EffectiveDataType.REAL);
-  private static final MessageTemplate[] SENDING_FIELD_TYPES = {
-    MessageTemplate.of("variables.elementaryWithType", MessageTemplate.of("variables.integer")),
-    MessageTemplate.of("variables.nonzeroInteger")
-  };
   List<Node> receivingFields;
```
Comment (lxpdd) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/statements/SetUpDownByStatement.java:29]: private

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/ElementaryProcessStandAlone.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  public void accept(ElementaryNode node, ProcessingContext ctx) {
+    ctx.getErrors().addAll(validatePicClauseUsage(node));
+    ctx.getErrors().addAll(validateCompatibleUsageClause(node));
+  }
+
+  private List<SyntaxError> validatePicClauseUsage(ElementaryNode node) {
+    if (StringUtils.isBlank(node.getPicClause())) {
+      return Collections.emptyList();
+    }
+    if (UsageFormatUtils.noPicClauseUsage.contains(node.getUsageFormat()))
```
Comment (lxpdd) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/ElementaryProcessStandAlone.java:49]: missing parentheses in this:  `{}`

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/ElementaryProcessStandAlone.java [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        ? MessageTemplate.of(BLANK_WHEN_ZERO)
+        : MessageTemplate.of(SIGN_CLAUSE);
+  }
+
+  private List<SyntaxError> validateCompatibleUsageClause(ElementaryNode node) {
+    if (Objects.isNull(node.getUsageFormat())) {
+      return Collections.emptyList();
+    }
+    boolean check = node.isSignClausePresent() || node.isBlankWhenZeroPresent();
+    if (UsageFormatUtils.blankAndSignNotAllowedUsage.contains(node.getUsageFormat()) && check)
```
Comment (lxpdd) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/ElementaryProcessStandAlone.java:71]: missing parentheses : `{`,  `}` for if

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/statements/SetToOnOffStatement.java (hunk 4 above)
Comment (ishche) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/statements/SetToOnOffStatement.java:29]: It's generated by lombok

**Pull request closed without merging**